File: 2021/07/04/main.go
package main

func countVowelPermutation(n int) int {
	a, e, i, o, u := 1, 1, 1, 1, 1

	for j := 1; j < n; j++ {
		a, e, i, o, u = e%1000000007, (a+i)%1000000007, (a+e+o+u)%1000000007, (i+u)%1000000007, a%1000000007
	}

	return (a + e + i + o + u) % 1000000007
}

File: 2021/07/04/main.py
class Solution:
    def countVowelPermutation(self, n: int) -> int:
        a = e = i = o = u = 1
        
        for _ in range(1, n):
            a,e,i,o,u = e, a+i, a+e+o+u, i+u, a
        
        return (a+e+i+o+u) % (10 ** 9 + 7)